### DISCUSSION TITLE
.RequestAccess() problem in IOS

### DISCUSSION THREAD

**Discussion by PXInformatica** [shinyorg/shiny#1521] [Q&A]
I have this code:
`var pushManager = Shiny.Hosting.Host.GetService<IPushManager>();
if (pushManager.RegistrationToken == null)
{
    var result = await pushManager.RequestAccess();
    if (result.Status != AccessState.Available)
    {
        //await this.Dialogs.Alert("Failed to register - " + result.Status);
    }
    else
    {
        //this.RegistrationToken = result.RegistrationToken;
    }
}
`
But on iOS, I don't get a response from:
`var result = await pushManager.RequestAccess();`

It simply doesn't do anything, I don't get any result. I'm doing this with Firebase, and I believe the configuration is correct because if I change the App ID to an incorrect one, I do get an error saying it doesn't match. On Android, I have no issues, since I can get the token.
What could be the problem?

**Comment by aritchie** [comment]
I have not updated to a new iOS binding that has support for Firebase.  You can implement your own and use the IPushProvider https://shinylib.net/client/push/providers/  - I will be updating this in v4 of Shiny, but that has no release timeline at the moment.
